src/backends/backend/python/python_interpreter.cc

```diff
@@ -0,0 +1,902 @@
+      model_config(std::move(model_config))
+{
+}
+
+TRITONSERVER_Error*
+ModelInstanceState::Create(
+    ModelState* model_state, TRITONBACKEND_ModelInstance* triton_model_instance,
+    ModelInstanceState** state)
+{
+  const char* instance_name;
+  RETURN_IF_ERROR(
+      TRITONBACKEND_ModelInstanceName(triton_model_instance, &instance_name));
+
+  TRITONSERVER_InstanceGroupKind instance_kind;
+  RETURN_IF_ERROR(
+      TRITONBACKEND_ModelInstanceKind(triton_model_instance, &instance_kind));
+
+  int32_t instance_id;
+  RETURN_IF_ERROR(
+      TRITONBACKEND_ModelInstanceDeviceId(triton_model_instance, &instance_id));
+
+  TRITONBACKEND_Model* triton_model;
+  RETURN_IF_ERROR(
+      TRITONBACKEND_ModelInstanceModel(triton_model_instance, &triton_model));
+
+  // TODO: Remove version 1
+  TRITONSERVER_Message* config_message;
+  RETURN_IF_ERROR(TRITONBACKEND_ModelConfig(triton_model, 1, &config_message));
+
+  // Parse JSON config
+  const char* buffer;
+  size_t byte_size;
+  RETURN_IF_ERROR(
+      TRITONSERVER_MessageSerializeToJson(config_message, &buffer, &byte_size));
+
+  ni::TritonJson::Value model_config;
+
+  TRITONSERVER_Error* err = model_config.Parse(buffer, byte_size);
+  RETURN_IF_ERROR(TRITONSERVER_MessageDelete(config_message));
+  RETURN_IF_ERROR(err);
+
+  *state = new ModelInstanceState(
+      model_state, triton_model_instance, instance_name, instance_kind,
+      instance_id, std::move(model_config));
+
+  return nullptr;
+}
+
+ModelInstanceState::~ModelInstanceState()
+{
+  // Close python interpreter.
+  grpc::ClientContext context;
+  ni::StatusCode result;
+  ni::Empty null_msg;
+
+  const auto err = stub->InterpreterShutdown(&context, null_msg, &result);
+  if (!err.ok() || result.code() != 0) {
+    LOG_ERROR << "Cannot shutdown interpreter gracefully: "
+              << err.error_message() << std::endl;
+  }
+
+  int status;
+  waitpid(interpreter_pid_, &status, 0);
+  unlink(domain_socket_.substr(7).c_str());
+}
+
+int
+ModelInstanceState::GetInputTensor(
+    const uint32_t iidx, TRITONBACKEND_Request* request,
+    ni::InferenceData* input_tensor,
+    std::vector<TRITONBACKEND_Response*>& responses, size_t r,
+    uint32_t& batch_size)
+{
+  const char* input_name;
+  // Load iidx'th input name
+  RESPOND_AND_RETURN_IF_ERROR(
+      request, TRITONBACKEND_RequestInputName(request, iidx, &input_name));
+
+  // Load iidx'th input
+  TRITONBACKEND_Input* in;
+  RESPOND_AND_RETURN_IF_ERROR(
+      request, TRITONBACKEND_RequestInput(request, input_name, &in));
+
+  // Load input properties
+  TRITONSERVER_DataType input_dtype;
+  const int64_t* input_shape;
+  uint32_t input_dims_count;
+  uint64_t input_byte_size;
+  uint32_t input_buffer_count;
+  GUARDED_RESPOND_IF_ERROR(
+      responses, r,
+      TRITONBACKEND_InputProperties(
+          in, &input_name, &input_dtype, &input_shape, &input_dims_count,
+          &input_byte_size, &input_buffer_count));
+
+  if (iidx == 0) {
```

Comment:
No, you can't do this in general to find batch size. Also you don't need to know the batch size.